### PR TITLE
[Facilities] Artwork content type 

### LINKED ISSUE #7026: Facilities Artwork Content Type
# Story

As a Facilities site owner, I would like our content editors to have the ability to create artwork content so that our users will eventually be able to browse a filterable gallery, see artwork on buildings and see works of art by a particular artist.

# Additional Info

Seems specific to this site.

**Content type create notes**
https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/Content%20type%20creation%20notes/Content%20type%20creation%20notes.aspx

They would like to keep their same url pattern otherwise they will need to update the urls on their maps https://uiadmin.maps.arcgis.com/apps/webappviewer/index.html?id=c4bcf56619a241deb3e8f490ce1b9ed6. Their current pattern is `art-on-campus/artwork/[node:title]`

# Question

Should https://github.com/uiowa/uiowa/issues/7028 be done as part of this or separate slice?

# Proposed Solution

### LINKED ISSUE #7028: artwork fields and design
**Current Fields**

- [x] Title
- [x] Media - Currently allows unlimited images and video but looks like every piece of artwork has one image and some have one video. The images don't all have the same aspect ratio. (e.g. https://www.facilities.uiowa.edu/art-on-campus/artwork/diapason, https://www.facilities.uiowa.edu/art-on-campus/artwork/2-3-1-1). Teaser images all render the same size currently.
- [x] Date - Year only, integer text field currently
- [x] Artist - entity reference to a artist content type currently
- [x] Building - reference an existing Facilities building - currently select list
- [x] Location - long text text area restricted to plain text currently
- [x] Medium - term reference to tag type taxonomy, allows multiple, gallery allows filtering by medium.
- [x] Dimension - text field currently
- [x] Website - Requires title if there is a url currently but seems like they are always outputting "Artist's Website" in the title field. Single url field
- [x] Status - single value, required field. Currently select list with following options (Uninstalled, Active, Not Currently On View)
- [x] Description - body? long text text area using filtered html text format
- [ ] Notes - no longer needed
- [ ] Geo - no longer needed

**Design**

Current artwork node - https://www.facilities.uiowa.edu/art-on-campus/artwork/voyage
Mockups - #7042

### LINKED ISSUE #7042: Artwork mockup for full node and teaser
# Story

As a web team, we would like to put together some mockups of an artwork node and teaser so we can get feedback/approval on design and to help us determine how we should structure this.

# Additional Info

Example node from current site https://www.facilities.uiowa.edu/art-on-campus/artwork/voyage

Media - Currently allows unlimited images and video but looks like every piece of artwork has one image and some have one video. The images don't all have the same aspect ratio. (e.g. https://www.facilities.uiowa.edu/art-on-campus/artwork/diapason, https://www.facilities.uiowa.edu/art-on-campus/artwork/2-3-1-1). Teaser images all render the same size currently.

# Proposed Solution

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uiowa/issues/7026
Resolves https://github.com/uiowa/uiowa/issues/7028
Resolves https://github.com/uiowa/uiowa/issues/7042

# How to test
```
ddev blt frontend && ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local uli /node/add/artwork
```

## Test content type creation
Confirm that it lines up to the standards set on https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/Content%20type%20creation%20notes/Content%20type%20creation%20notes.aspx?
- Schema blueprints are not set
- Moderation workflow not set

### Full node
Recreate a full node of https://www.facilities.uiowa.edu/art-on-campus/artwork/iowa-portraits
Confirm pathauto pattern is set. 

**Image**

- Use focal point to set on the face of the portrait
- Confirm that widescreen crop image follows the focal point set

### Teaser

- Create a quick artwork teaser view and confirm that the teaser looks like https://github.com/uiowa/uiowa/issues/7042#issuecomment-1783373741.

### CODE REVIEW

**Comment by bspeare** [comment]
Moving back to draft to make a few updates before the meeting today at 1pm.

**Comment by cory-skeers** [comment]
Any thoughts on nonstandard entries for the year?
image

It accepts valid integers, but since there's no min or max, you can enter things like year -13 or year 19999999. 

Personally, I feel like it's a content thing (nothing breaks if you have a negative, for instance), and at best adding any sort of constraints might catch a random typo (like the 19996 in my example), but is extra work/complexity for not much benefit imo.

Wanted to throw it out there if there are other opinions, or if we had something in place already elsewhere we want to adapt here too.

**Comment by bspeare** [comment]
@cory-skeers I'd probably trust them to fix it.  Is there anything simple you'd recommend on the field settings for that?

**Comment by pyrello** [comment]
> Any thoughts on nonstandard entries for the year?

I think that we should leave this as a content issue. If there is rampant cases of the non-year values being put in, they could maybe make the case to use why we should add some validation for them.

**Comment by cory-skeers** [comment]
> @cory-skeers I'd probably trust them to fix it. Is there anything simple you'd recommend on the field settings for that?
> I think that we should leave this as a content issue. If there is rampant cases of the non-year values being put in, they could maybe make the case to use why we should add some validation for them.

Thanks! Those match my thinking on it, but I wanted to leave the door open for other thoughts after I saw it. Officially closing that door now.

**Comment by cory-skeers** [comment]
Do we want the option to add a new artist from the Artwork edit form? (Similar to how you can add a new author or point of contact from an Article edit form.)

**Comment by cory-skeers** [comment]
Should we add Artwork to be able to be referenced in Featured Content blocks? (I added an additional note in https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/Content%20type%20creation%20notes/Content%20type%20creation%20notes.aspx -- I think we'll probably want to ask this of all new pieces of content)

**Comment by joewhitsitt** [comment]
> Should we add Artwork to be able to be referenced in Featured Content blocks? (I added an additional note in https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/Content%20type%20creation%20notes/Content%20type%20creation%20notes.aspx -- I think we'll probably want to ask this of all new pieces of content)

I think that is a good question, but at the same time, how often has this been asked for after the fact? I see we did it for ITS service but that was to avoid building a custom view.

**Comment by cory-skeers** [comment]
> > Should we add Artwork to be able to be referenced in Featured Content blocks? (I added an additional note in https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/Content%20type%20creation%20notes/Content%20type%20creation%20notes.aspx -- I think we'll probably want to ask this of all new pieces of content)
> 
> I think that is a good question, but at the same time, how often has this been asked for after the fact? I see we did it for ITS service but that was to avoid building a custom view.

I'm not aware of any ask to have things included. Maybe leave it off (for this and other new content types) unless we know a specific use case and/or it gets asked for down the line? Helps avoid an extra partial split, even if it's a small one.

**Comment by briand44** [comment]
As far as artwork being used in Featured Content.... They do have a use case where this could be used (Recent Installations on https://facilities.prod.drupal.uiowa.edu/campus-spaces/art-campus), but when I probed them on this it didn't sound like those change often and they didn't feel like they needed that capability.

**Comment by briand44** [comment]
>Do we want the option to add a new artist from the Artwork edit form? (Similar to how you can add a new author or point of contact from an Article edit form.)

I don't have strong feelings about this. I don't know how useful it is if we only expose a few artist fields (e.g. first and last name). I think it would make things easier for them if they were able to add an artist and all the relevant field info but they don't currently have that ability on their existing site and they haven't asked for that yet.

**Comment by pyrello** [comment]
> As far as artwork being used in Featured Content.... They do have a use case where this could be used (Recent Installations on https://facilities.prod.drupal.uiowa.edu/campus-spaces/art-campus), but when I probed them on this it didn't sound like those change often and they didn't feel like they needed that capability.

Adding it to featured content would essentially just be allow for them to have a consistently formatted teaser card experience if they wanted to highlight certain curated artwork. Featured content is curated, so they would have to change it if they wanted it to change. Seems like it could be a good fit if its a small change to make it work.

**Comment by cory-skeers** [comment]
> > As far as artwork being used in Featured Content.... They do have a use case where this could be used (Recent Installations on https://facilities.prod.drupal.uiowa.edu/campus-spaces/art-campus), but when I probed them on this it didn't sound like those change often and they didn't feel like they needed that capability.
> 
> Adding it to featured content would essentially just be allow for them to have a consistently formatted teaser card experience if they wanted to highlight certain curated artwork. Featured content is curated, so they would have to change it if they wanted it to change. Seems like it could be a good fit if its a small change to make it work.

I went ahead and added it, given the small size. https://facilities.prod.drupal.uiowa.edu/campus-spaces/art-campus might still end up being a more custom built page with the content that's showcased on there now, but I could see there being similar spots where they'd want a quick and easy way to highlight some of the content.

> > Do we want the option to add a new artist from the Artwork edit form? (Similar to how you can add a new author or point of contact from an Article edit form.)
> 
> I don't have strong feelings about this. I don't know how useful it is if we only expose a few artist fields (e.g. first and last name). I think it would make things easier for them if they were able to add an artist and all the relevant field info but they don't currently have that ability on their existing site and they haven't asked for that yet.

Maybe we can re-address in #7113, and if anything done there might affect what fields we would want shown (or logic to set the person type to artist if created here, etc.) in case that work makes anything apparent.

File: docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/Entity/Artwork.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      if (!empty($artist_entity) && $artist_entity->getType() === 'person') {
+        // Access the title or name of the referenced "person" entity directly.
+        // Use 'label()' method to get the title/name.
+        $artist_name = $artist_entity->label();
+
+        $year_value = $year_field->getString();
+
+        if (!empty($artist_name) && !empty($year_value)) {
+          $subtitle = $artist_name . ', ' . $year_value;
+          $build['#subtitle'] = $subtitle;
```
Comment (cory-skeers) [inline docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/Entity/Artwork.php:44]: Do we know if there is or will be any art that doesn't have an associated artist? If so, the year won't be displayed for the teaser view. (Year is required, so the alternative can't happen, where we have an artist but no year)

Comment (cory-skeers) [reply docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/Entity/Artwork.php:44]: > Do we know if there is or will be any art that doesn't have an associated artist? If so, the year won't be displayed for the teaser view. (Year is required, so the alternative can't happen, where we have an artist but no year)

I checked the existing site, and there is one (and only one) artist for every artwork node, so I think we're fine here.

**Review by cory-skeers — APPROVED** [review]
Made a note to possibly revisit the artist reference field when that issue is worked on, and updated to include artwork in the featured content block.

Everything else checks out, and looks great!

**Comment by bspeare** [comment]
Thanks for the review @cory-skeers!

**Pull request merged**